ironfish/src/workerPool/tasks/createTransaction.ts

```diff
@@ -24,39 +28,210 @@ export type CreateTransactionRequest = {
       hashOfSibling: Buffer
     }[]
   }[]
-  receives: { publicAddress: string; amount: bigint; memo: string }[]
-}
+  readonly receives: { publicAddress: string; amount: bigint; memo: string }[]
 
-export type CreateTransactionResponse = {
-  type: 'createTransaction'
-  serializedTransactionPosted: Uint8Array
-}
+  constructor(
+    spendKey: string,
+    transactionFee: bigint,
+    expirationSequence: number,
+    spends: {
+      note: Buffer
+      treeSize: number
+      rootHash: Buffer
+      authPath: { side: Side; hashOfSibling: Buffer }[]
+    }[],
+    receives: { publicAddress: string; amount: bigint; memo: string }[],
+    jobId?: number,
+  ) {
+    super(WorkerMessageType.CreateTransaction, jobId)
+    this.spendKey = spendKey
+    this.transactionFee = transactionFee
+    this.expirationSequence = expirationSequence
+    this.spends = spends
+    this.receives = receives
+  }
+
+  serialize(): Buffer {
+    const bw = bufio.write(this.getSize())
+    bw.writeVarString(this.spendKey)
+    bw.writeVarBytes(BigIntUtils.toBytesBE(this.transactionFee))
+    bw.writeU64(this.expirationSequence)
+    bw.writeU64(this.spends.length)
+
+    for (const spend of this.spends) {
+      bw.writeVarBytes(Buffer.from(spend.note))
```

Comment:
Does this work?

```suggestion
      bw.writeVarBytes(spend.note)
```

Reply:
It seems that the tests in `accounts.test.slow.ts` fail if `spend.note` is not explicitly a `Buffer`.